drivers/adc/ad7124/ad7124.c

```diff
@@ -439,170 +417,105 @@ int32_t ad7124_read_data(struct ad7124_dev *dev,
 	/* Get the read result */
 	*p_data = regs[AD7124_Data].value;
 
-	return ret;
+	return 0;
 }
 
-/**
+/***************************************************************************//**
  * @brief Get the ID of the channel of the latest conversion.
- *
  * @param dev     - The handler of the instance of the driver.
  * @param status  - Pointer to store the read data.
- *
- * @return Returns 0 for success or negative error code.
- */
+ * @return Returns 0 for success or negative error code otherwise.
+*******************************************************************************/
 int32_t ad7124_get_read_chan_id(struct ad7124_dev *dev, uint32_t *status)
 {
 	int32_t ret;
 	uint32_t reg_temp;
 
 	ret = ad7124_read_register2(dev, AD7124_STATUS_REG, &reg_temp);
-	if (ret != 0)
+	if (ret)
 		return ret;
+
 	*status = reg_temp & AD7124_STATUS_REG_CH_ACTIVE(0xF);
 
-	return ret;
+	return 0;
 }
 
 /***************************************************************************//**
  * @brief Computes the CRC checksum for a data buffer.
- *
  * @param p_buf    - Data buffer
  * @param buf_size - Data buffer size in bytes
- *
  * @return Returns the computed CRC checksum.
 *******************************************************************************/
 uint8_t ad7124_compute_crc8(uint8_t * p_buf, uint8_t buf_size)
 {
 	uint8_t i = 0;
 	uint8_t crc = 0;
 
-	while(buf_size) {
-		for(i = 0x80; i != 0; i >>= 1) {
+	while (buf_size) {
+		for (i = 0x80; i != 0; i >>= 1) {
 			bool cmp1 = (crc & 0x80) != 0;
 			bool cmp2 = (*p_buf & i) != 0;
-			if(cmp1 != cmp2) { /* MSB of CRC register XOR input Bit from Data */
+			if (cmp1 != cmp2) {
```

Comment:
this is a general rule: if the `if` statement has brackets, then also the `else` statement should have brackets. Please revert this change.